Sources/PostgresNIO/New/Extensions/Logging+PSQL.swift

```diff
@@ -0,0 +1,246 @@
+        case messageID = "psql_message_id"
+        case messagePayload = "psql_message_payload"
+        
+        
+        case database = "psql_database"
+        case username = "psql_username"
+        
+        case userEvent = "psql_user_event"
+    }
+}
+
+@usableFromInline
+struct PostgresLoggingMetadata: ExpressibleByDictionaryLiteral {
+    @usableFromInline
+    typealias Key = PSQLConnection.LoggerMetaDataKey
+    @usableFromInline
+    typealias Value = Logger.MetadataValue
+    
+    @usableFromInline var _baseRepresentation: Logger.Metadata
+    
+    @usableFromInline
+    init(dictionaryLiteral elements: (PSQLConnection.LoggerMetaDataKey, Logger.MetadataValue)...) {
+        let values = elements.lazy.map { (key, value) -> (String, Self.Value) in
+            (key.rawValue, value)
+        }
+        
+        self._baseRepresentation = Logger.Metadata(uniqueKeysWithValues: values)
+    }
+    
+    @usableFromInline
+    subscript(postgresLoggingKey loggingKey: PSQLConnection.LoggerMetaDataKey) -> Logger.Metadata.Value? {
+        get {
+            return self._baseRepresentation[loggingKey.rawValue]
+        }
+        set {
+            self._baseRepresentation[loggingKey.rawValue] = newValue
+        }
+    }
+    
+    @inlinable
+    var representation: Logger.Metadata {
+        self._baseRepresentation
+    }
+}
+
+
+extension Logger {
+    
+    static let psqlNoOpLogger = Logger(label: "psql_do_not_log", factory: { _ in SwiftLogNoOpLogHandler() })
+    
+    @usableFromInline
+    subscript(postgresMetadataKey metadataKey: PSQLConnection.LoggerMetaDataKey) -> Logger.Metadata.Value? {
+        get {
+            return self[metadataKey: metadataKey.rawValue]
+        }
+        set {
+            self[metadataKey: metadataKey.rawValue] = newValue
+        }
+    }
+    
+}
+
+extension Logger {
+    
+    /// Log a message passing with the `Logger.Level.trace` log level.
+    ///
+    /// If `.trace` is at least as severe as the `Logger`'s `logLevel`, it will be logged,
+    /// otherwise nothing will happen.
+    ///
+    /// - parameters:
+    ///    - message: The message to be logged. `message` can be used with any string interpolation literal.
+    ///    - metadata: One-off metadata to attach to this log message
+    ///    - source: The source this log messages originates to. Currently, it defaults to the folder containing the
+    ///              file that is emitting the log message, which usually is the module.
+    ///    - file: The file this log message originates from (there's usually no need to pass it explicitly as it
+    ///            defaults to `#file`).
+    ///    - function: The function this log message originates from (there's usually no need to pass it explicitly as
+    ///                it defaults to `#function`).
+    ///    - line: The line this log message originates from (there's usually no need to pass it explicitly as it
+    ///            defaults to `#line`).
```

Comment:
Do we really need a repeat of all of the documentation for every one of `Logger`'s logging methods? Tanner and I tend to use a style like this in these scenarios:
```swift
/// See `Logger.trace(_:metadata:source:file:function:line:)`
```